### PR TITLE
Document use of mamba to install Cylc.

### PR DESCRIPTION
**Pull request by hjoliver**
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/installation.rst [hunk 1]
```diff
@@ -15,15 +15,22 @@ Cylc runs on Unix-like systems including Linux and Mac OS.
...
+.. warning::
+
+   The ``conda`` environment solver can be very slow and memory-intensive for
+   complex packages. We recommend installing the much faster ``mamba`` solver
+   first, and then installing Cylc with that.
+
 .. code-block:: sub
 
-   $ conda install -c conda-forge cylc-flow
+   $ conda install -c conda-forge mamba
```
Comment (oliver-sanders) [inline src/installation.rst:26]: This suggests installing `mamba` into the cylc env (or cylc into the base env).

**Review by oliver-sanders — COMMENTED** [review]
The `mamba` solver is now available in `conda` too.

* https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community
* https://conda.github.io/conda-libmamba-solver/getting-started/#getting-started

I think it still requires explicit installation/configuration but that may depend on where you got conda from.

Maybe replace this warning with a `.. note::` just saying "We recommend using `mamba`".

Mamba is the free as in speaking-chips replacement for Anaconda's free as in free until-you-cross-the-billing-threshold implementation which will hopefully supplant conda in the near future.

File: src/installation.rst (hunk 1 above)
Comment (hjoliver) [reply src/installation.rst:26]: I see I have an extra erroneous line in there (fixed) ... but it is OK to install mamba into the cylc env.

**Comment by hjoliver** [comment]
> The mamba solver is now available in conda too.

Nice, I wasn't aware of that.

**Comment by hjoliver** [comment]
@oliver-sanders - I've tweaked the text along the lines you suggested. See what you think.

File: src/installation.rst (hunk 1 above)
Comment (oliver-sanders) [reply src/installation.rst]: The docs says conda and mamba will only work when installed in the "base" env. You can install them into other envs but I guess they won't work for environment management, dunno.

**Review by oliver-sanders — APPROVED** [review]
Cheers.

**Pull request merged**